tests/unit/Mvc/ModelCest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Phalcon\Test\unit\Mvc;
+
+use Phalcon\Cache\Backend\Apc;
+use Phalcon\Cache\Frontend\Data;
+use Phalcon\Test\Models\Robots;
+use UnitTester;
+
+/**
+ * \Phalcon\Test\Unit\Mvc\ModelCest
+ * Tests the Phalcon\Mvc\Model component
+ *
+ * @copyright (c) 2011-2016 Phalcon Team
+ * @link      http://www.phalconphp.com
+ * @author    Andres Gutierrez <andres@phalconphp.com>
+ * @author    Serghei Iakovlev <serghei@phalconphp.com>
+ * @author    Wojciech Ślawski <jurigag@gmail.com>
+ * @package   Phalcon\Test\Unit\Mvc
+ *
+ * The contents of this file are subject to the New BSD License that is
+ * bundled with this package in the file docs/LICENSE.txt
+ *
+ * If you did not receive a copy of the license and are unable to obtain it
+ * through the world-wide-web, please send an email to license@phalconphp.com
+ * so that we can send you a copy immediately.
+ */
+class ModelCest
+{
+    public function testModelsCacheSnapshot(UnitTester $I)
+    {
+        $I->wantToTest("Saving snapshot while using modelsCache");
+
+        if (!extension_loaded('apc')) {
+            throw new \PHPUnit_Framework_SkippedTestError(
+                'Warning: apc extension is not loaded'
+            );
+        }
+
+        if (!ini_get('apc.enabled') || (PHP_SAPI === 'cli' && !ini_get('apc.enable_cli'))) {
+            throw new \PHPUnit_Framework_SkippedTestError(
+                'Warning: apc.enable_cli must be set to "On"'
+            );
+        }
+
+        if (extension_loaded('apcu') && version_compare(phpversion('apcu'), '5.1.6', '=')) {
+            throw new \PHPUnit_Framework_SkippedTestError(
+                'Warning: APCu v5.1.6 was broken. See: https://github.com/krakjoe/apcu/issues/203'
+            );
+        }
+
+        $I->addServiceToContainer(
+            'modelsCache',
+            function () {
+                return new Apc(new Data(['lifetime' => 20]));
```

Comment:
@Jurigag Could you please use another, more standard cache driver here to avoid checks for APC?